common/elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/ElasticsearchIndexCreator.scala

```diff
@@ -23,18 +16,29 @@ class ElasticsearchIndexCreator(
   config: IndexConfig)(implicit ec: ExecutionContext)
     extends Logging {
 
-  def create: Future[Unit] = {
-    create(
-      index = index,
-      mapping = config.mapping,
-      config.analysis,
-      config.searchTemplate)
+  def create: Future[Unit] = createOrUpdate
+
+  val mapping = config.mapping
+  val analysis = config.analysis
+  val searchTemplate = config.searchTemplate
+
+  private def exists = elasticClient.execute(indexExists(index.name))
+
+  private def createOrUpdate: Future[Unit] = {
+    for {
+      existsResp <- exists
+      createResp <- if (existsResp.result.isExists) {
+        update
+      } else {
+        put
+      }
+      resp <- searchTemplate
+        .map(template => putSearchTemplate(template))
+        .getOrElse(Future.successful(createResp))
+    } yield { handleEsError(resp) }
   }
```

Comment:
Might suggest this as a slightly clearer version

```suggestion
  private def exists =
    elasticClient.execute(indexExists(index.name)).map(_.result.isExists)

  private def createOrUpdate: Future[Unit] = {
    for {
      doesExist <- exists
      createResp <- if (doesExist) update else put
      resp <- searchTemplate
        .map(putSearchTemplate)
        .getOrElse(Future.successful(createResp))
    } yield { handleEsError(resp) }
  }
```

Reply:
I suppose I find it just a little weird to return a `Future[Boolean]` as all the others futures are of `Future[Response[T]]` but that doesn't matter as we're not using `EitherT`.

Thanks!